netbox/netbox/models/features.py

```diff
@@ -84,6 +85,15 @@ def to_objectchange(self, action):
         by ChangeLoggingMiddleware.
         """
         from extras.models import ObjectChange
+
+        postchange_data = None
```

Comment:
This probably isn't the ideal place to hook in, because it requires modifying all the overridden calls to `to_objectchange()` inside individual models with additional logic.

Instead, we wrap the call to `save()` in the `handle_changed_object()` signal receiver, to ensure the ObjectChange instance is saved only if the pre- and post-change data differ. To further simplify this, we can add a property on ObjectChange which returns essentially `self.prechange_data == self.postchange_data`.